lib/index.js

```diff
@@ -75,33 +85,19 @@ internals.Connection.prototype.start = function (callback) {
 
     // Create client
 
-    var server = new MongoDB.Server(this.settings.host, this.settings.port, { auto_reconnect: true, poolSize: this.settings.poolSize });
-    this.client = new MongoDB.Db(this.settings.partition, server, { safe: true });
+    this.client = MongoDB.MongoClient;
```

Comment:
I think, you don't need to keep reference to MongoClient, its not instance. Basically just keeping reference to `db` once its instantiated via connect is sufficient.